velox/expression/Expr.cpp

```diff
@@ -234,7 +235,11 @@ void Expr::computeMetadata() {
   // a special form, and all its inputs are also deterministic.
   deterministic_ = isCurrentFunctionDeterministic();
```

Comment:
should we fix isCurrentFunctionDeterministic() for LambdaExpr instead? We can then avoid special handling of LambdaExpr in L238

Reply:
@bikramSingh91 Discussed offline and decided to keep this PR short. Bikram may follow-up with more changes.